lib/TermUtil.ts

```diff
@@ -85,7 +85,7 @@ export function getLiteralLanguage(literalValue: string): string {
  * @param {RDF.DataFactory} dataFactory An optional datafactory to create terms with.
  * @return {RDF.Term} An RDF-JS term.
  */
-export function stringToTerm(value: string, dataFactory?: RDF.DataFactory): RDF.Term {
+export function stringToTerm(value: string, dataFactory?: RDF.DataFactory<RDF.BaseQuad>): RDF.Term {
```

Comment:
Could you add `<Q extends RDF.BaseQuad = RDF.Quad>` as generic type to this function here as well? Otherwise non-`RDF.BaseQuad` factories will not be usable here.

Reply:
Every factory will have an implementation that extends `RDF.BaseQuad`? (I didn't add a generic here since it's not needed.)

Reply:
I'm not sure if TS sees `RDF.DataFactory<SomeExtensionOfBaseQuad>` as an extension of `RDF.DataFactory<RDF.BaseQuad>`.
I know that Java does not considers this to be valid, so I would expect TS to also not allow this.
Perhaps a test can be added for it to make sure?

Reply:
TS is fine with it: pushed https://github.com/rubensworks/rdf-string.js/pull/14/commits/545e15887b473ac8f31906f770bc76a1adbae987.

Reply:
Ok, great, TIL :-)